x-pack/legacy/plugins/lens/public/indexpattern_plugin/dimension_panel/bucket_nesting_editor.tsx

```diff
@@ -32,35 +35,75 @@ export function BucketNestingEditor({
   const columns = Object.entries(layer.columns);
   const aggColumns = columns
     .filter(([id, c]) => id !== columnId && c.isBucketed)
-    .map(([value, c]) => ({ value, text: c.label }));
+    .map(([value, c]) => ({
+      value,
+      text: c.label,
+      fieldName: hasField(c) ? c.sourceField : '',
+    }));
 
   if (!column || !column.isBucketed || !aggColumns.length) {
     return null;
   }
 
+  const fieldName = hasField(column) ? column.sourceField : '';
+
   const prevColumn = layer.columnOrder[layer.columnOrder.indexOf(columnId) - 1];
 
   if (aggColumns.length === 1) {
     const [target] = aggColumns;
 
+    function toggleNesting() {
+      if (prevColumn) {
+        setColumns(nestColumn(layer.columnOrder, columnId, target.value));
+      } else {
+        setColumns(nestColumn(layer.columnOrder, target.value, columnId));
+      }
+    }
+
     return (
       <>
         <EuiHorizontalRule margin="m" />
-        <EuiSwitch
-          data-test-subj="indexPattern-nesting-switch"
-          label={i18n.translate('xpack.lens.xyChart.nestUnderTarget', {
-            defaultMessage: 'Nest under {target}',
-            values: { target: target.text },
+        <EuiFormRow
+          label={i18n.translate('xpack.lens.indexPattern.groupingControlLabel', {
+            defaultMessage: 'Grouping',
           })}
-          checked={!!prevColumn}
-          onChange={() => {
-            if (prevColumn) {
-              setColumns(nestColumn(layer.columnOrder, columnId, target.value));
-            } else {
-              setColumns(nestColumn(layer.columnOrder, target.value, columnId));
-            }
-          }}
-        />
+        >
+          <>
+            <EuiRadio
+              id={generator()}
```

Comment:
I'm not familiar with this, but is this predictive, or is it going to cause render churn? It seems totally fine to me to just specify an ID schematically here.

Reply:
Looks like I'm supposed to pass a string, and it'll generate consistent ids if given a string.